packages/fetchai/protocols/fipa/dialogues.py

```diff
@@ -335,18 +327,66 @@ def get_dialogue(self, fipa_msg: Message, agent_addr: Address) -> Dialogue:
             other_initiated_dialogue = cast(
                 FipaDialogue, self.dialogues[other_initiated_dialogue_label]
             )
-            if other_initiated_dialogue.is_valid_next_message(fipa_msg):
-                result = other_initiated_dialogue
+            result = other_initiated_dialogue
         if self_initiated_dialogue_label in self.dialogues:
             self_initiated_dialogue = cast(
                 FipaDialogue, self.dialogues[self_initiated_dialogue_label]
             )
-            if self_initiated_dialogue.is_valid_next_message(fipa_msg):
-                result = self_initiated_dialogue
+            result = self_initiated_dialogue
         if result is None:
             raise ValueError("Should have found dialogue.")
         return result
 
+    def update(
+        self,
+        message: Message,
+        is_self_message: bool,
```

Comment:
is_self_message? why is this needed

Reply:
We can infer it from the state of the dialogue..

Reply:
There is no dialogue here. 
The idea is, you've got a message here, either created by yourself or received from another agent. 
Then you want to update your dialogue models with this new message. You first have to check if it is a new message for a new dialogue. if it is, then you have to check whether the message is by you or your opponent and create the new dialogue accordingly (call `self_initiated` vs `other_initiated`). That's where we need the is_self_message, since we cannot infer it from the message. If messages had the sender address, we could retrieve this info and call the appropriate `create_dialogue` method.